Library/Ark/ArkPackage.cs

```diff
@@ -344,30 +346,55 @@ private string[] readFileNames(Stream header)
     }
 
     /// <summary>
-    /// Reads the new file table format in v10
+    /// Reads the new file table format in v9 and v10
     /// </summary>
-    private void readNewFileTable(Stream header)
+    private void readNewFileTable(Stream header, bool readHash = false)
     {
       uint numFiles = header.ReadUInt32LE();
       var files = new OffsetFile[numFiles];
 
-      for (var i = 0; i < numFiles; i++)
+      if (readHash)
       {
-        // Version 3 uses 32-bit file offsets
-        long arkFileOffset = header.ReadInt64LE();
-        string path = header.ReadLengthPrefixedString(System.Text.Encoding.UTF8);
-        var flags = header.ReadInt32LE();
-        uint size = header.ReadUInt32LE();
+        for (var i = 0; i < numFiles; i++)
+        {
+          // Version 3 uses 32-bit file offsets
+          long arkFileOffset = header.ReadInt64LE();
+          string path = header.ReadLengthPrefixedString(System.Text.Encoding.UTF8);
+          var flags = header.ReadInt32LE();
+          uint size = header.ReadUInt32LE();
+          header.ReadUInt32LE();
```

Comment:
It would be a bit less wasteful to just do `header.Position += 4` or `header.Seek(4, SeekOrigin.Current)` with a comment explaining that there's a checksum we're ignoring.

Reply:
No excuse for this one. I'll fix it immediately.